helix-term/src/commands.rs

```diff
@@ -429,13 +429,20 @@ pub fn replace(cx: &mut Context) {
             ..
         } = event
         {
-            let text = Tendril::from_char(ch);
-
             let (view, doc) = cx.current();
 
             let transaction =
                 Transaction::change_by_selection(doc.text(), doc.selection(view.id), |range| {
-                    (range.from(), range.to() + 1, Some(text.clone()))
+                    let max_to = doc.text().len_chars().saturating_sub(1);
+                    let to = std::cmp::min(max_to, range.to() + 1);
+                    let text: String = doc
+                        .text()
+                        .slice(range.from()..to)
+                        .chars()
+                        .map(|c| if c == '\n' { '\n' } else { ch })
+                        .collect();
+
+                    (range.from(), to, Some(text.as_str().into()))
```

Comment:
Do we need to convert to `String` first and then convert to `&str` and then to `Rope`? Why not just directly convert into `Rope` or `&str`?

Reply:
The parameter is an `Option<Tendril>`, a small string abstraction (uses stack for short lengths, will heap allocate if necessary). I think the `as_str()` call isn't needed here, `into()` might be enough.

Reply:
That's true, the `as_str` called can be omitted. I was pretty sure that it gave me a compilation error when I tried this before.